includes/class-wc-subscriptions-manager.php

```diff
@@ -868,26 +873,36 @@ public static function maybe_cancel_subscription( $post_id ) {
 	 * This function fixes that by setting '_wp_trash_meta_status' to 'wc-cancelled' whenever its former status
 	 * is something that can not be restored.
 	 *
-	 * @param int $post_id
+	 * @param int $id
 	 * @since 1.0.0 - Migrated from WooCommerce Subscriptions v2.0
 	 */
-	public static function fix_trash_meta_status( $post_id ) {
+	public static function fix_trash_meta_status( $id ) {
+
+		$subscription = wcs_get_subscription( $id );
+		if ( ! $subscription ) {
+			return;
+		}
+
+		// Check its type because $subscription might be WC_Subscription's child class.
```

Comment:
> WC_Subscription's child class.

I don't think we need to emphasize that this check is for custom subscription types. I don't think we need this comment but if we want to keep it, we can just say something like: 

```suggestion
		// Check the object type is shop_subscription
```

Reply:
Sounds good! Addressed in [6a1f79c](https://github.com/Automattic/woocommerce-subscriptions-core/pull/305/commits/6a1f79c17117d8599b8bd34f627c4252c9d5df34)